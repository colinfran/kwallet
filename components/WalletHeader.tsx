import React, { useContext } from "react"
import { StyleSheet, View, Dimensions } from "react-native"
import { Skeleton, Text, Heading, useContrastText } from "native-base"
import { DataContext } from "../providers/DataProvider"
import DoubleButton from "./Button/DoubleButton"
import { useNavigation } from "@react-navigation/native"
import { getLocales } from "expo-localization"
import { getCurrencySymbol } from "../constants/currencies"

const WalletAmount = (): JSX.Element => {
  const { graphData, appColor, backgroundColor, textColor, selectedCurrency } =
    useContext(DataContext)
  const navigation = useNavigation()

  const apiVal = graphData ? graphData : { currentPrice: "0" }

  const isLoaded = graphData

  // console.log(apiVal?.walletBalance)

  const walletTotal = apiVal?.walletBalance?.available || 0

  const { currentPrice = "0" } = apiVal
  const value = parseFloat(walletTotal) * parseFloat(currentPrice)

  const deviceLanguage = getLocales()[0]
  const locale = `${
    deviceLanguage.languageCode
  }-${deviceLanguage.measurementSystem.toUpperCase()}`

  const currencyFormatter = (val: number): string => {
    const symb = getCurrencySymbol(selectedCurrency)
    const formatter = new Intl.NumberFormat(locale, {
      style: "currency",
      currency: selectedCurrency,
      currencyDisplay: "symbol",
    })
    const str = formatter.format(val).toString().split(symb)[1]
    return `${symb}${str}`
  }

  const walletTotalMonetaryValue = currencyFormatter(value)

  const textColorPressed = useContrastText(appColor)
  return (
    <View
      style={[
        styles.container,
        {
          backgroundColor: backgroundColor,
          // borderColor: appColor,
          // shadowColor: textColor,
        },
      ]}
    >
      <View style={{ height: 60 }}>
        <Skeleton isLoaded={isLoaded} marginBottom={2} w={310}>
          <Heading color={textColor} textAlign={"center"} size="3xl" width={Dimensions.get("window").width} testID="test-walletAmountUSD">
            {walletTotalMonetaryValue}
          </Heading>
        </Skeleton>
      </View>
      <View>
        <Skeleton isLoaded={isLoaded} w={230}>
          <Text color={textColor} fontSize="xl" testID="test-walletAmount">
            {`${walletTotal} KAS`}
          </Text>
        </Skeleton>
      </View>
    </View>
  )
}

export default WalletAmount

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center",
  },
  shadow: {
    shadowOffset: {
      width: 0,
      height: 5,
    },
    shadowOpacity: 0.35,
    shadowRadius: 3.5,
    elevation: 5,
  },
})
