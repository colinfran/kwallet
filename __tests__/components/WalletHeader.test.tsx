import React from "react"
import WalletHeader from "../../components/WalletHeader"
import { Providers } from "../helper"
import { render, screen } from "@testing-library/react-native"

const mockedDispatch = jest.fn()

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native")
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: mockedDispatch,
    }),
  }
})

describe("<WalletHeader />", () => {
  beforeEach(() => {
    mockedDispatch.mockClear()
  })
  it("renders correctly", () => {
    const tree = render(<WalletHeader />, { wrapper: Providers }).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("buttons", () => {
    render(<WalletHeader />, { wrapper: Providers })
    const leftBtn = screen.getByTestId("test-left")
    const rightBtn = screen.getByTestId("test-right")
    expect(leftBtn).toHaveTextContent("Send")
    expect(rightBtn).toHaveTextContent("Receive")
  })

  it("wallet amount", () => {
    render(<WalletHeader />, { wrapper: Providers })
    const amount = screen.getByTestId("test-walletAmount")
    const amountUsd = screen.getByTestId("test-walletAmountUSD")
    expect(amount).toHaveTextContent("0 KAS")
    expect(amountUsd).toHaveTextContent("$0.00")
  })
})
