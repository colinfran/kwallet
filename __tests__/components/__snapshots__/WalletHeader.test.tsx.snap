// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WalletHeader /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "elevation": 5,
          "justifyContent": "center",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.5,
        },
        {},
      ]
    }
  >
    <View>
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#000",
            "fontFamily": undefined,
            "fontSize": 48,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 2.4000000000000004,
            "lineHeight": 60,
            "textDecorationLine": undefined,
          }
        }
        testID="test-walletAmountUSD"
      >
        $0.00
      </Text>
    </View>
    <View>
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#000",
            "fontFamily": undefined,
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 30,
            "textDecorationLine": undefined,
          }
        }
        testID="test-walletAmount"
      >
        0 KAS
      </Text>
    </View>
    <View
      style={
        {
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 30,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 15,
              "justifyContent": "center",
              "width": "100%",
            },
            {},
          ]
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 40,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                dataSet={{}}
                pointerEvents="auto"
                style={
                  [
                    {
                      "borderColor": {
                        "100": "#fee2e2",
                        "200": "#fecaca",
                        "300": "#fca5a5",
                        "400": "#f87171",
                        "50": "#fef2f2",
                        "500": "#ef4444",
                        "600": "#dc2626",
                        "700": "#b91c1c",
                        "800": "#991b1b",
                        "900": "#7f1d1d",
                      },
                      "borderRadius": 15,
                      "flexDirection": "row",
                    },
                    [
                      {
                        "elevation": 5,
                        "shadowOffset": {
                          "height": 5,
                          "width": 0,
                        },
                        "shadowOpacity": 0.35,
                        "shadowRadius": 3.5,
                      },
                      {
                        "backgroundColor": "#654321",
                        "shadowColor": "#000",
                      },
                    ],
                  ]
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  colorScheme="blue"
                  dataSet={{}}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  pointerEvents="auto"
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#654321",
                        "borderBottomLeftRadius": 15,
                        "borderColor": "#404040",
                        "borderRadius": 0,
                        "borderTopLeftRadius": 15,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "paddingBottom": 10,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 10,
                        "width": "50%",
                      },
                      {
                        "borderColor": "#123456",
                        "borderWidth": 2,
                      },
                    ]
                  }
                  testID="test-left"
                >
                  <View
                    dataSet={{}}
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                    test={true}
                  >
                    <View
                      dataSet={{}}
                      style={{}}
                    >
                      <Text
                        dataSet={{}}
                        style={
                          {
                            "backgroundColor": undefined,
                            "color": "#000",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        Send
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  dataSet={{}}
                  style={
                    {
                      "width": 0,
                    }
                  }
                />
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  colorScheme="blue"
                  dataSet={{}}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  pointerEvents="auto"
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#654321",
                        "borderBottomRightRadius": 15,
                        "borderColor": "#404040",
                        "borderLeftWidth": 0,
                        "borderRadius": 0,
                        "borderTopRightRadius": 15,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "paddingBottom": 10,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 10,
                        "width": "50%",
                      },
                      {
                        "borderColor": "#123456",
                        "borderWidth": 2,
                      },
                    ]
                  }
                  testID="test-right"
                >
                  <View
                    dataSet={{}}
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                    test={true}
                  >
                    <View
                      dataSet={{}}
                      style={{}}
                    >
                      <Text
                        dataSet={{}}
                        style={
                          {
                            "backgroundColor": undefined,
                            "color": "#000",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        Receive
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
